mountpoint-s3/src/cli.rs

```diff
@@ -511,9 +519,46 @@ impl CliArgs {
         filesystem_config.cache_config = self.cache_config();
         filesystem_config.mem_limit = self.mem_limit();
         filesystem_config.use_upload_checksums = self.should_use_upload_checksum(s3_personality);
+
+        if let Some(username) = &self.run_as_user {
+            match Self::resolve_user_group(username) {
+                Ok((uid, gid)) => {
+                    filesystem_config.uid = uid; // Sets filesystem operations to run as this user
+                    filesystem_config.gid = gid;
+                    tracing::info!("Running Mountpoint as user: {}", username);
+
+                    // Convert to proper types for setuid/setgid
+                    let gid = Gid::from_raw(gid);
```

Comment:
Perhaps we could make `resolve_user_group` return Gid/Uid and do `to_raw` once each rather than doing a back and forth